rs/ledger_suite/icrc1/ledger/src/main.rs

```diff
@@ -301,23 +372,25 @@ fn encode_metrics(w: &mut ic_metrics_encoder::MetricsEncoder<Vec<u8>>) -> std::i
             ledger.blockchain().num_archived_blocks.saturating_add(ledger.blockchain().blocks.len() as u64) as f64,
             "Total number of transactions stored in the main memory, plus total number of transactions sent to the archive.",
         )?;
-        let token_pool: Nat = ledger.balances().token_pool.into();
-        w.encode_gauge(
-            "ledger_balances_token_pool",
-            token_pool.0.to_f64().unwrap_or(f64::INFINITY),
-            "Total number of Tokens in the pool.",
-        )?;
-        let total_supply: Nat = ledger.balances().total_supply().into();
-        w.encode_gauge(
-            "ledger_total_supply",
-            total_supply.0.to_f64().unwrap_or(f64::INFINITY),
-            "Total number of tokens in circulation.",
-        )?;
-        w.encode_gauge(
-            "ledger_balance_store_entries",
-            ledger.balances().store.len() as f64,
-            "Total number of accounts in the balance store.",
-        )?;
+        if is_ready() {
```

Comment:
Should we provide any metrics while the ledger is not ready?

Reply:
They could be useful while debugging an upgrade with migration - you can see the heap/stable memory usage, number of blocks, etc.

Reply:
Sure, but we must be careful to not access something that is not available (and would trap the entire message). Are there tests for fetching metrics during migration?

Reply:
I added a test: https://github.com/dfinity/ic/pull/1513/commits/6fed6f302c8382fb42a106e6883dca3bd72ce67d